packages/react-magma-dom/src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,321 @@
+            setShowLabelsLayout(true);
+          } else if (layout === StepperLayout.hideLabels) {
+            setHideLabelsLayout(true);
+          } else if (layout === StepperLayout.summaryView) {
+            setSummaryViewLayout(true);
+          }
+        }
+        if (showLabelsLayout) {
+          setHideLabelsLayout(false);
+          setSummaryViewLayout(false);
+        } else if (hideLabelsLayout) {
+          setShowLabelsLayout(false);
+          setSummaryViewLayout(false);
+        } else if (summaryViewLayout) {
+          setShowLabelsLayout(false);
+          setHideLabelsLayout(false);
+        }
+      }
+
+      window.addEventListener('resize', handleResize);
+      handleResize();
+
+      return () => window.removeEventListener('resize', handleResize);
+    }, [windowWidth]);
+
+    // Step states
+    const step = React.Children.toArray(children);
+
+    const Steps = React.Children.map(
+      children,
+      (
+        child: React.ReactElement<React.PropsWithChildren<StepProps>>,
+        index
+      ) => {
+        if (child.type === Step) {
+          const stepStatusStyles =
+            currentStep >= index + 1
+              ? StepStatus.complete
+              : currentStep >= index
+              ? StepStatus.active
+              : StepStatus.incomplete;
+
+          const item = React.cloneElement(child, {
+            'aria-current': currentStep === index,
+            children: currentStep === index ? child : null,
+            key: index,
+            isInverse: isInverse,
+            hasHideLabels: hideLabelsLayout,
+            hasSummaryView: summaryViewLayout,
+            stepStatus: stepStatusStyles,
+          });
+
+          return (
+            <>
+              {item}
+              {index !== step.length - 1 && (
+                <StyledSeparator
+                  isInverse={isInverse}
+                  stepStatus={stepStatusStyles}
+                  theme={theme}
+                />
+              )}
+            </>
+          );
+        }
+      }
+    );
+
+    // When summaryView is set to true, this shows one step label and description at a time based on the active step below the Stepper component.
+
+    const StepLabels = React.Children.map(
+      children,
+      (
+        child: React.ReactElement<React.PropsWithChildren<StepProps>>,
+        index
+      ) => {
+        const item = child as React.ReactElement<
+          React.PropsWithChildren<StepProps>
+        >;
+
+        if (
+          item.type === Step &&
+          currentStep >= index &&
+          currentStep <= index
+        ) {
+          return child;
+        }
+      }
+    );
+
+    // Final step description
+    const completionLabel = props.completionLabel
+      ? props.completionLabel
+      : i18n.stepper.completionLabel;
+
+    // Shows both the step count out of total steps and the step label, allows for customization of the step title.
+    const stepSummary = (
```

Comment:
Since this is not reused and you're not doing any logic, I would prefer to not have it in a constant. For legibility purposes, lets throw the JSX in line 315 directly.